plugin/environment.go

```diff
@@ -439,6 +440,31 @@ func (env *Environment) RunMultiPluginHook(hookRunnerFunc func(hooks Hooks) bool
 	})
 }
 
+func (env *Environment) AddPluginEventSubscription(event string, plugin string) {
+	pluginSubscription, ok := env.pluginEventSubscriptions.Load(event)
+
+	if ok {
+		pluginSubscriptionList := pluginSubscription.([]string)
+		pluginSubscription = append(pluginSubscriptionList, plugin)
+	} else {
+		pluginSubscription = []string{plugin}
```

Comment:
I'm wondering if we want to make this into a defined `[]struct{}`. We only have a use case for one field right now so it is fine as a string, as we can easily change later since the map is ephemeral.

We will want to avoid duplicate plugin ids in the subscription map. I'm thinking that a given plugin would call `SubscribePluginEvent()` in its `OnActivate()` hook method. This may cause problems if the plugin is disabled/re-enabled and is already subscribed to the event. With that in mind I think we should be removing the subscription whenever a subscribed plugin is deactivated.